payment/p2tr.go

```diff
@@ -0,0 +1,170 @@
+package payment
+
+import (
+	"errors"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/vulpemventures/go-elements/address"
+	"github.com/vulpemventures/go-elements/network"
+	"github.com/vulpemventures/go-elements/taproot"
+)
+
+const (
+	segwitVersion = byte(0x01)
+)
+
+// TaprootPaymentData is included in Payment struct to store Taproot-related data
+type TaprootPaymentData struct {
+	XOnlyTweakedKey    []byte
```

Comment:
From my POV user should be able to access fields but not to mutate them(which is done with factory methods), this way we make sure pkg is not misused.

Reply:
factories, as the name suggests, are intended to be constructors for exported/unexported types.

The manipulation of the fields can be done via type's exported methods, or directly by the user if the inner fields are exported.

This is a library and its purpose is to give the user the building blocks to create any kind of application.
For this reason, I do think it's better to allow the user to make mistakes instead of him not being able to use the lib because it misses methods to do some very fancy stuff.

After all, all the rest of the library is designed to achieve this goal and there doesn't seem to be any valid reason to change it starting from now.